pkgs/tools/misc/adrgen/default.nix

```diff
@@ -0,0 +1,28 @@
+{ lib, buildGoModule, fetchFromGitHub }:
+
+buildGoModule rec {
+  pname = "adrgen";
+  version = "0.4.0-beta";
+
+  src = fetchFromGitHub {
+    owner = "asiermarques";
+    repo = "adrgen";
+    rev = "v${version}";
+    sha256 = "sha256-2ZE/orsfwL59Io09c4yfXt2enVmpSM/QHlUMgyd9RYQ=";
+  };
+
+  vendorSha256 = "sha256-aDtUD+KKKSE0TpSi4+6HXSBMqF/TROZZhT0ox3a8Idk=";
+
+  doInstallCheck = true;
+
+  installCheckPhase = ''
+    $out/bin/adrgen version
+  '';
```

Comment:
This should rather be a package test https://nixos.org/manual/nixpkgs/stable/#ssec-inline-package-tests-writing

Reply:
And there is [`testVersion`](https://github.com/NixOS/nixpkgs/blob/de49d559382ea05a304589b6fc13c9be68351810/pkgs/build-support/trivial-builders.nix#L724-L755) if you are interested in that. See https://github.com/NixOS/nixpkgs/commit/255c2849248e331d7f0e56a041b80e0014a19177 for how to use it.

Reply:
Thanks! I didn't know about these, I'll use `testVersion` for future packages as well :)